core/modules/qdisp/QueryRequest.h

```diff
@@ -50,7 +51,7 @@ class BadResponseError : public std::exception {
     BadResponseError(std::string const& s_)
         : std::exception(),
           s("BadResponseError:" + s_) {}
-    ~BadResponseError() throw() {}
+    virtual ~BadResponseError() throw() {}
     virtual char const* what() throw() {
```

Comment:
exact signature for this is:

    virtual const char* what() const noexcept;

(const is certainly missing here), same for RequestError class